packages/app/src/navigation/SidebarNavigation.cy.tsx

```diff
@@ -86,4 +116,74 @@ describe('SidebarNavigation', () => {
     mountComponent()
     cy.findByTestId('sidebar-link-specs-page').should('have.class', 'router-link-exact-active')
   })
+
+  context('debug status badge', () => {
+    it('renders new badge without cloudProject', { viewportWidth: 1280 }, () => {
+      cy.clock(new Date('December 20, 2022'))
+
+      mountComponent()
+
+      cy.findByLabelText('New Debug feature').should('be.visible').contains('New')
+      cy.percySnapshot('Debug Badge:collapsed')
+
+      cy.findByLabelText(defaultMessages.sidebar.toggleLabel.collapsed, {
+        selector: 'button',
+      }).click()
+
+      cy.percySnapshot('Debug Badge:expanded badge')
+    })
+
+    it('renders new badge when run status is "NOTESTS" or "RUNNING"', () => {
+      cy.clock(new Date('December 20, 2022'))
+
+      for (const status of ['NOTESTS', 'RUNNING'] as CloudRunStatus[]) {
+        mountComponent({ cloudProject: { status, numFailedTests: 0 } })
+        cy.findByLabelText('New Debug feature').should('be.visible').contains('New')
+      }
+    })
+
+    it('renders no badge if no cloudProject and released > 2 months ago', () => {
+      // Set to February 15, 2023 to see this fail
+      cy.clock(new Date('February 20, 2023'))
+      mountComponent()
+      cy.findByLabelText('New Debug feature').should('not.exist')
+    })
+
+    it('renders success badge when status is "PASSED"', () => {
+      mountComponent({ cloudProject: { status: 'PASSED', numFailedTests: 0 } })
+      cy.findByLabelText('Last recorded run passed').should('be.visible').contains('0')
+      cy.percySnapshot('Debug Badge:passed')
+    })
+
+    it('renders failure badge', () => {
+      mountComponent({ cloudProject: { status: 'FAILED', numFailedTests: 1 } })
+      cy.findByLabelText('Last recorded run had 1 test failure').should('be.visible').contains('1')
+      cy.percySnapshot('Debug Badge:failed')
+
+      mountComponent({ cloudProject: { status: 'FAILED', numFailedTests: 10 } })
+      cy.findByLabelText('Last recorded run had 10 test failures').should('be.visible').contains('9+')
+      cy.percySnapshot('Debug Badge:failed:truncated')
+    })
+
+    it('renders failure badge when failing tests and abnormal status', () => {
+      for (const status of ['CANCELLED', 'ERRORED', 'OVERLIMIT', 'TIMEDOUT'] as CloudRunStatus[]) {
+        mountComponent({ cloudProject: { status, numFailedTests: 4 } })
+        cy.findByLabelText('Last recorded run had 4 test failures').should('be.visible').contains('4')
+      }
+    })
+
+    it('renders error badge when no tests and abnormal status', () => {
+      for (const status of ['CANCELLED', 'ERRORED', 'OVERLIMIT', 'TIMEDOUT'] as CloudRunStatus[]) {
+        mountComponent({ cloudProject: { status, numFailedTests: 0 } })
+        cy.findByLabelText('Last recorded run had an error').should('be.visible').contains('0')
+      }
+
+      cy.percySnapshot('Debug Badge:errored')
+    })
+
```

Comment:
Now we've got Vite 4, we should definitely get Vitest in - testing this complex logic via components makes reviewing a bit challenging. It looks like you've covered all the cases, though, which is great.

Reply:
We might have to steal some of the gql-testing infra if we implement vitest as this logic is heavily tied to gql

Reply:
I think we could probably decouple the logic from the UI - and just keep the component tests to make sure it's all wired up correctly. Anyway, another battle for another day.